src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -993,7 +1009,7 @@ private void WriteUpdateMethod(Node node)
 
             string wasUpdatedCheck(Field field)
             {
-                var format = field.Type.EndsWith("TypeSymbol") ? "!TypeSymbol.Equals({0}, this.{1}, TypeCompareKind.ConsiderEverything)" : "{0} != this.{1}";
+                var format = field.Type.StartsWith("TypeSymbol") ? "!TypeSymbol.Equals({0}, this.{1}, TypeCompareKind.ConsiderEverything)" : "{0} != this.{1}";
```

Comment:
>StartsWith("TypeSymbol") [](start = 40, length = 24)

Should this be `TrimEnd('?') == "TypeSymbol"`?

Reply:
Yeah, probably. I thought of a better way to do it later in the PR 😄